crates/task-impls/src/consensus.rs

```diff
@@ -1100,20 +1110,63 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     self.current_proposal = None;
                 }
             }
-            HotShotEvent::VidCertRecv(cert) => {
-                debug!("VID cert received for view ! {}", *cert.view_number);
+            HotShotEvent::VidDisperseRecv(disperse, sender) => {
+                let view = disperse.data.get_view_number();
 
-                let view = cert.get_view_number();
-                self.vid_certs.insert(view, cert);
+                debug!(
+                    "VID disperse received for view: {:?} in consensus task",
+                    view
+                );
 
-                // RM TODO: VOTING
-            }
+                // stop polling for the received disperse
+                self.quorum_network
+                    .inject_consensus_info(ConsensusIntentEvent::CancelPollForVIDDisperse(
+                        *disperse.data.view_number,
+                    ))
+                    .await;
```

Comment:
I think we should only stop polling for VIDDisperse after we've verified that this is a valid share.  So I think we should move this block of code further down in the function.